tests/unit/test_state.py

```diff
@@ -0,0 +1,178 @@
+    PATH_CONFIG_NAME,
+    PROTOCOL_CONFIG_NAME,
+    Configuration,
+)
+
+
+def test_valid_config():
+    """
+    arrange: Mock charm with valid configurations.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+
+    config = Configuration.from_charm(charm)
+    assert config.hostname == "example.com"
+    assert config.path == "/"
+    assert config.backends == (IPv4Address("10.10.1.1"), IPv4Address("10.10.2.2"))
+    assert config.protocol == "https"
+
+
+def test_empty_hostname():
+    """
+    arrange: Mock charm with empty hostname.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[HOSTNAME_CONFIG_NAME] = "   "
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+    assert (
+        str(err.value) == "Config error: ['hostname = : String should have at least 1 character']"
+    )
+
+
+def test_long_hostname():
+    """
+    arrange: Mock charm with long hostname.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[HOSTNAME_CONFIG_NAME] = "a" * 256
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "Value error, Hostname cannot be longer than 255" in str(err.value)
+
+
+def test_invalid_hostname():
+    """
+    arrange: Mock charm with hostname with invalid character.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[HOSTNAME_CONFIG_NAME] = "example?.com"
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert "consist of alphanumeric and hyphen" in str(err.value)
+
+
+def test_empty_path():
+    """
+    arrange: Mock charm with empty path.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PATH_CONFIG_NAME] = "   "
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+    assert str(err.value) == "Config error: ['path = : String should have at least 1 character']"
+
+
+def test_invalid_path():
+    """
+    arrange: Mock charm with path with invalid character.
+    act: Create the configuration from the charm.
+    assert: Correct configurations from the mock charm.
+    """
+    charm = MockCharmFactory()
+    charm.config[PATH_CONFIG_NAME] = "/^"
+
+    with pytest.raises(ConfigurationError) as err:
+        Configuration.from_charm(charm)
+
+    assert (
+        str(err.value)
+        == "Config error: ['path = /^: Value error, Path contains non-allowed character']"
+    )
+
```

Comment:
I'd add a test with a more complex path than "/" something like "/path/to/destination"

Reply:
Added a test.